fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/common/StoreTimerSnapshot.java

```diff
@@ -0,0 +1,186 @@
+
+/*
+ * StoreTimerSnapshot.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2018 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.common;
+
+import com.apple.foundationdb.annotation.API;
+import com.apple.foundationdb.record.logging.KeyValueLogMessage;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * An immutable snapshot of a {@link StoreTimer}.
+ *
+ * Subtracting a snapshot of a timer from the original timer after subsequent operations have
+ * been performed and timed can provide useful metrics as to the cost of those subsequent operations.
+ * See {@link StoreTimer#subtractSnapshot(StoreTimer, StoreTimerSnapshot)}.
+ */
+@API(API.Status.MAINTAINED)
+public final class StoreTimerSnapshot {
+    @Nonnull
+    private final Map<StoreTimer.Event, StoreTimerSnapshot.CounterSnapshot> counters;
```

Comment:
Maybe this was an artifact of your IDE, but it is kind of strange to prefix `CounterSnapshot` with `StoreTimerSnapshot` when this class is `StoreTimerSnapshot`.  In other words, I would think this would be:

    private final Map<StoreTimer.Event, CounterSnapshot> counters;